stdlib/abc.pyi

```diff
@@ -1,26 +1,31 @@
 from _typeshed import SupportsWrite
-from typing import Any, Callable, Dict, Tuple, Type, TypeVar
+from typing import Any, Callable, Generic, Tuple, Type, TypeVar
+from typing_extensions import Concatenate, ParamSpec
 
 _T = TypeVar("_T")
+_ClsT = TypeVar("_ClsT", bound=Type[Any])
+_P = ParamSpec("_P")
+_R_co = TypeVar("_R_co", covariant=True)
 _FuncT = TypeVar("_FuncT", bound=Callable[..., Any])
 
 # These definitions have special processing in mypy
 class ABCMeta(type):
     __abstractmethods__: frozenset[str]
-    def __init__(self, name: str, bases: Tuple[type, ...], namespace: Dict[str, Any]) -> None: ...
+    def __init__(self, name: str, bases: Tuple[type, ...], namespace: dict[str, Any]) -> None: ...
     def __instancecheck__(cls: ABCMeta, instance: Any) -> Any: ...
     def __subclasscheck__(cls: ABCMeta, subclass: Any) -> Any: ...
     def _dump_registry(cls: ABCMeta, file: SupportsWrite[str] | None = ...) -> None: ...
     def register(cls: ABCMeta, subclass: Type[_T]) -> Type[_T]: ...
 
 def abstractmethod(funcobj: _FuncT) -> _FuncT: ...
 
-class abstractproperty(property): ...
+class abstractclassmethod(classmethod[_ClsT, _P, _R_co], Generic[_ClsT, _P, _R_co]):
+    def __init__(self, callable: Callable[Concatenate[_ClsT, _P], _R_co]) -> None: ...
 
-# These two are deprecated and not supported by mypy
-def abstractstaticmethod(callable: _FuncT) -> _FuncT: ...
-def abstractclassmethod(callable: _FuncT) -> _FuncT: ...
+class abstractstaticmethod(staticmethod[_FuncT], Generic[_FuncT]):
+    def __init__(self, callable: _FuncT) -> None: ...
```

Comment:
Is there any other difference than the name of the argument (`callable` here, `f` in builtins)? It seems that both should be keyword-only:

```
Python 3.8.5 (default, May 27 2021, 13:30:53) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> staticmethod(f=print)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: staticmethod() takes no keyword arguments
>>> from abc import abstractstaticmethod
>>> abstractstaticmethod(f=print)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: __init__() got an unexpected keyword argument 'f'
```

Reply:
Just noticed that the error messages are different. Seems like the current `abstractstaticmethod` annotation is correct, but the built-in `staticmethod` is positional-only.

Reply:
Same for `@classmethod`. I'm changing it.